module/zfs/zvol.c

```diff
@@ -821,14 +817,14 @@ zvol_request(struct request_queue *q, struct bio *bio)
 		 * Some requests are just for flush and nothing else.
 		 */
 		if (uio.uio_resid == 0) {
-			if (bio->bi_rw & VDEV_REQ_FLUSH)
+			if (bio_is_flush(bio))
 				zil_commit(zv->zv_zilog, ZVOL_OBJ);
 			goto out2;
 		}
 
 		error = zvol_write(zv, &uio,
-		    ((bio->bi_rw & (VDEV_REQ_FUA|VDEV_REQ_FLUSH)) ||
-		    zv->zv_objset->os_sync == ZFS_SYNC_ALWAYS));
+		    bio_is_flush(bio) || bio_is_fua(bio) ||
```

Comment:
Here we don't need REQ_PREFLUSH, though I'm not sure about REQ_OP_FLUSH, since from document it seems we won't get REQ_OP_FLUSH.